crates/uv-cli/src/lib.rs

```diff
@@ -3766,6 +3766,28 @@ pub struct ToolInstallArgs {
     #[arg(long, value_delimiter = ',', value_parser = parse_maybe_file_path)]
     pub with_requirements: Vec<Maybe<PathBuf>>,
 
+    /// Constrain versions using the given requirements files.
+    ///
+    /// Constraints files are `requirements.txt`-like files that only control the _version_ of a
+    /// requirement that's installed. However, including a package in a constraints file will _not_
+    /// trigger the installation of that package.
+    ///
+    /// This is equivalent to pip's `--constraint` option.
+    #[arg(long, short, alias = "constraint", env = EnvVars::UV_CONSTRAINT, value_delimiter = ' ', value_parser = parse_maybe_file_path)]
+    pub constraints: Vec<Maybe<PathBuf>>,
+
+    /// Override versions using the given requirements files.
+    ///
+    /// Overrides files are `requirements.txt`-like files that force a specific version of a
+    /// requirement to be installed, regardless of the requirements declared by any constituent
+    /// package, and regardless of whether this would be considered an invalid resolution.
+    ///
+    /// While constraints are _additive_, in that they're combined with the requirements of the
+    /// constituent packages, overrides are _absolute_, in that they completely replace the
+    /// requirements of the constituent packages.
+    #[arg(long, alias = "override", env = EnvVars::UV_OVERRIDE, value_delimiter = ' ', value_parser = parse_maybe_file_path)]
```

Comment:
Don't we want to be able to use `--override` for non-file overrides? e.g., `--override foo==0.1.0`

(same for `--constraint`)

Reply:
I don’t think we allow that elsewhere, but we could? How would we differentiate between file and non-file on the CLI?

Reply:
I think `--override` and `--overrides` would be the distinguishing factor.

I'm honestly not sure what the status quo is outside `uv pip`. I think there's a tracking issue about this somewhere, but not sure where...

Reply:
I guess in the long-term `--override` / `--overrides` vs `--override-file` / `--overrides-file` might make more sense? It's sort of awkward since we have the precedents from the pip interface.

Reply:
To be dealt with later :)